simple_rpc/simple_rpc.py

```diff
@@ -44,6 +44,35 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
+    def _open(self):
+        if self.is_open():
+            return
+        try:
+            self._connection.open()
+        except SerialException as error:
+            raise IOError(error.strerror.split(':')[0])
+
+    def _close(self):
+        if not self.is_open():
+            return
+        self._connection.close()
+
+    def _auto_open(f):
+        """Decorator for automatic opening and closing of ethernet sockets."""
+        def _auto_open_wrapper(self, *args, **kwargs):
+            if self._is_socket and not self.is_open():
+                self._open()
+
+            result = f(self, *args, **kwargs)
+
+            if self._is_socket:
+                self._close()
```

Comment:
Here we call the private function `_close`, which does nothing but closing the port.

Reply:
ok